### PR TITLE
Add `HighestSlotStatesBelow` DB getter

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements `HighestSlotStatesBelow` DB getter and tests. 

Last piece of DB feature to be used by new state mgmt and gen service  🙌

### CODE REVIEW

File: beacon-chain/db/kv/state.go [hunk 1]
```diff
@@ -383,6 +384,46 @@ func (k *Store) HighestSlotStates(ctx context.Context) ([]*state.BeaconState, er
 	return states, nil
 }
 
+// statesAtSlotBitfieldIndex retrieves the states in DB given the input index. The index represents
+// the position of the slot bitfield the saved state maps to.
+func (k *Store) statesAtSlotBitfieldIndex(ctx context.Context, tx *bolt.Tx, index uint64) ([]*state.BeaconState, error) {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.statesAtSlotBitfieldIndex")
+	defer span.End()
+
+	highestSlot := index - 1
```
Comment (nisdas) [inline beacon-chain/db/kv/state.go]: what if index is 0, this overflows

Comment (terencechain) [reply beacon-chain/db/kv/state.go]: addressed the line below :) 

`highestSlot = uint64(math.Max(0, float64(highestSlot)))`

Comment (nisdas) [reply beacon-chain/db/kv/state.go]: wouldn't 
```
math.Max(0, float64(highestSlot))
```
return the larger number which will be `MaxFloat64` ?

Comment (terencechain) [reply beacon-chain/db/kv/state.go]: Nice catch! It's now taking int instead of uint64, also verified via a test. Thank you :)

**Pull request merged**